internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -231,6 +234,10 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 			return fmt.Errorf("error creating siderolink spec: %w", err)
 		}
 
+		if err = ctrl.clearSiderolinkAddresses(ctx, r); err != nil {
```

Comment:
we can pair on that, but I feel it might be cleaner, and more controller-like, if we introduce a formal `for{}` loop here which will set up the connection, check its status, and re-setup if the connection is down

Reply:
Got it, you mean also without failing the controller to reconnect, right?

Reply:
well, to some extent. if the Wireguard connection goes down, it's not a failure, we should just re-setup and inject new endpoint/address.

Also one more thing I forgot to mention - when cleaning up `AddressSpec` resources, and it should probably clean them up always, it should only clean up its own (where `Metadata().Owner() == ctrl.Name()`)

Reply:
@smira It seems the owner of the `AddressSpec` is `AddressMergeController` here - not sure what to do about it.

Reply:
oh right 🤦 always forget about that.

Reply:
done